webrender/src/clip_scroll_node.rs

```diff
@@ -520,7 +520,8 @@ impl ClipScrollNode {
                     state.nearest_scrolling_ancestor_viewport
                        .translate(&info.origin_in_parent_reference_frame);
 
-                if !info.resolved_transform.preserves_2d_axis_alignment() {
+                if !info.resolved_transform.preserves_2d_axis_alignment() ||
+                   info.resolved_transform.has_perspective_component() {
```

Comment:
Hrm...does this work well with 90 degree rotations?

Reply:
Ah, good point! I updated the patch to do a full relative transform calculation and transform the rect. That code path almost never gets hit in reality, so the performance difference should be negligible.